### PR TITLE
Use ha-dialog for dialog-system-log-details, add close icon button

### PR DESCRIPTION
**Pull request by joshmcrty**
## Proposed change

This replaces the use of `ha-paper-dialog` with `ha-dialog` for the System Log detail dialog on the System Log page. It also adds a close icon button so the dialog can be closed on mobile devices.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

Click on an entry in the Configuration > Log page to view the dialog.

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #6138
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/logs/dialog-system-log-detail.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import "@material/mwc-icon-button/mwc-icon-button";
-import { mdiContentCopy } from "@mdi/js";
+import { mdiContentCopy, mdiClose } from "@mdi/js";
 import "@polymer/paper-dialog-scrollable/paper-dialog-scrollable";
```
Comment (bramkragten) [inline src/panels/config/logs/dialog-system-log-detail.ts:3]:
```suggestion
```

File: src/panels/config/logs/dialog-system-log-detail.ts [hunk 2]
```diff
@@ -69,13 +68,9 @@ class DialogSystemLogDetail extends LitElement {
...
-      <ha-paper-dialog
-        with-backdrop
-        opened
-        @opened-changed="${this._openedChanged}"
-      >
-        <div class="heading">
-          <h2>
+      <ha-dialog open @closed=${this.closeDialog} hideActions heading=${true}>
+        <div slot="heading" class="heading">
+          <h2 class="title">
```
Comment (bramkragten) [inline src/panels/config/logs/dialog-system-log-detail.ts:73]: use `createCloseHeading` https://github.com/home-assistant/frontend/blob/dev/src/panels/config/users/dialog-user-detail.ts#L65-L66

Comment (ludeeus) [reply src/panels/config/logs/dialog-system-log-detail.ts:73]: Can't do that with the copy icon being there as well?

Comment (bramkragten) [reply src/panels/config/logs/dialog-system-log-detail.ts:73]: So I should actually expand the stuff Github hides for me to see what the rest of the code is... 🤦

File: src/panels/config/logs/dialog-system-log-detail.ts [hunk 3]
```diff
@@ -69,13 +68,9 @@ class DialogSystemLogDetail extends LitElement {
...
     return html`
-      <ha-paper-dialog
-        with-backdrop
-        opened
-        @opened-changed="${this._openedChanged}"
-      >
-        <div class="heading">
-          <h2>
+      <ha-dialog open @closed=${this.closeDialog} hideActions heading=${true}>
+        <div slot="heading" class="heading">
```
Comment (bramkragten) [inline src/panels/config/logs/dialog-system-log-detail.ts:72]: Use the `<ha-header-bar>`:

```ts
          <ha-header-bar>
            <span slot="title">
              ${this.hass.localize(
	              "ui.panel.config.logs.details",
	              "level",
	              item.level
	          )}
            </span>
	          <mwc-icon-button id="copy" @click=${this._copyLog} slot="actionItems">
	            <ha-svg-icon .path=${mdiContentCopy}></ha-svg-icon>
	          </mwc-icon-button>
	          <paper-tooltip
	            manual-mode
	            for="copy"
	            position="left"
	            animation-delay="0"
	            offset="4"
	            >${this.hass.localize("ui.common.copied")}</paper-tooltip
	          >
	          <mwc-icon-button slot="actionItems" dialogAction="cancel">
	            <ha-svg-icon .path=${mdiClose}></ha-svg-icon>
	          </mwc-icon-button>
          </ha-header-bar>
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**